pydap_extras/handlers/pcic.py

```diff
@@ -0,0 +1,325 @@
+        os.remove(f.name)
+        return response
+
+    def create_ini(self, sesh, net_name, native_id):
+        """Creates the actual text of a pydap SQL handler config file and returns it as a StringIO. `self.filepath` should be set before this is called. It will typically be something like ``.../[network_name]/[native_id].rsql``. The database station_id is looked up from that.
+
+        :param environ: WSGI environment which *must* contain a dsn string under the key pydap.handlers.pcic.dsn
+        :rtype: StringIO.StringIO
+        """
+        q = (
+            sesh.query(Station.id)
+            .join(Network)
+            .filter(Station.native_id == native_id)
+            .filter(Network.name == net_name)
+        )
+        if not q.first():
+            raise ValueError(
+                "No such station {net_name}/{native_id}".format(**locals())
+            )
+        (station_id,) = q.first()
+
+        full_query = self.get_full_query(station_id, sesh)
+
+        q = (
+            sesh.query(
+                Station.native_id,
+                History.station_name,
+                Network.name,
+                History.the_geom,
+                History.elevation,
+            )
+            .join(History)
+            .join(Network)
+            .filter(Station.id == station_id)
+        )
+        if q.count() < 1:
+            native_id, station_name, network, lat, lon, elevation = (
+                native_id,
+                "",
+                net_name,
+                float("nan"),
+                float("nan"),
+                float("nan"),
+            )
+        else:
+            if q.count() > 1:
+                logger.warning(
+                    "Multiple history entries (ids {}) were found for a single station_id, but we're reporting locations for the most recent".format(
+                        []
+                    )
+                )
+                (sdate,) = (
+                    sesh.query(func.max(History.sdate))
+                    .filter(History.station_id == station_id)
+                    .first()
+                )
+                # This should never happen
+                if not sdate:
+                    raise ValueError(
+                        "Found multiple history entries for station_id {}, but none have a valid record start date!".format(
+                            station_id
+                        )
+                    )
+                q = q.filter(History.sdate == sdate)
+
+            _, station_name, network, geom, elevation = q.first()
+            elevation = elevation if elevation else float("nan")
+            lat, lon = (
+                (sesh.scalar(ST_Y(geom)), sesh.scalar(ST_X(geom)))
+                if geom is not None
+                else (float("nan"), float("nan"))
+            )
+
+        dsn = self.dsn
+        full_query = full_query.replace('"', '\\"')
+
+        s = (
+            """database:
```

Comment:
Since this is now just for Python 3, maybe we could use fstrings here and ditch the old fashioned percent string formatting?